File: santefluxpatientduval-duval_version/OptifluxApp/src/app/models/contrat.ts
export interface WorkDay {
    day: string;
    start_time: string;
    end_time: string;
}
  
export interface Contrat {
    id?: string;
    user_id: string;
    speciality: string;
    contrat_type: string;
    contrat_hours: string;
    work_days: WorkDay[];
}

File: santefluxpatientduval-duval_version/OptifluxAPI/schemas/contrat.py
from pydantic import BaseModel
from typing import List

class WorkDay(BaseModel):
    day: str
    start_time: str  # Format: "HH:MM", ex. "09:00"
    end_time: str    # Format: "HH:MM", ex. "17:00"

class ContratCreate(BaseModel):
    user_id: str
    speciality: str
    contrat_type: str
    contrat_hours: str
    work_days: List[WorkDay]  # Liste d'objets WorkDay
    created_at: str = None
    updated_at: str = None
    matricule: str = None